### DISCUSSION TITLE
reload config plugin go.d/ping without restart service netdata

### DISCUSSION THREAD

**Discussion by cayenne17** [netdata/netdata#16063] [Q&A] [answered]
Hello,
I would like to know if there is a way of adding a go.d/ping host and reloading the configuration without restarting netdata and without having any interruption in the graphics?

**Comment by ilyam8** [comment]
Hey, @cayenne17. You can reload go.d.plugin by sending SIGHUP to its process PID. You will get small gaps only on go.d.plugin charts.

**Reply by ilyam8** [reply, marked as answer]
Alternatively, you can try using the file-watching feature (go.d.plugin rereads it on every change and adds/stops jobs automatically) - but it is not documented and could break in the future. But if you want:

- create a file anywhere and make sure it is readable by `netdata` user.
- add `-w PATH_TO_THE_FILE` to go.d.plugin command options in `netdata.conf`.
- ...
- profit!

Example:

- create `/opt/netdata/etc/netdata/ping.conf` (could be `/etc/netdata/ping.conf` in your case, depends on the install prefix)

```yaml
jobs:
  - name: first
    hosts:
      - 8.8.8.8
```

- make go.d.plugin to watch it

```
# netdata.conf
[plugin:go.d]
        # update every = 1
        command options = -w /opt/netdata/etc/netdata/ping.conf
```

- restart netdata service

After that, you can add/remove jobs in `/opt/netdata/etc/netdata/ping.conf`, go.d.plugin will pick up the changes automatically.

**Comment by ilyam8** [comment]
@cayenne17 Also we are working on a feature that will allow you to do the same (add/remove jobs, check their status, etc.) using UI.
